recipes/mursst/recipe.py

```diff
@@ -0,0 +1,184 @@
+        'secret': creds['secretAccessKey'],
+        'token': creds['sessionToken'],
+        'anon': False,
+    }
+
+
+def earthdata_auth(username: str, password: str):
+    if earthdata_protocol == 's3':
+        return get_s3_creds(username, password)
+    else:
+        token = get_earthdata_token(username, password)
+        return {'headers': {'Authorization': f'Bearer {token}'}}
+
+
+fsspec_open_kwargs = earthdata_auth(ED_USERNAME, ED_PASSWORD)
+
+
+@dataclass
+class FilterVars(beam.PTransform):
+    """Filter kerchunk variables by name."""
+
+    keep: Set[str] = field(default_factory=set)
+
+    @staticmethod
+    def _filter(item: list[dict], keep: Set[str]) -> list[dict]:
+        filtered = []
+        for translated in item:
+            # see Kerchunk docs for dropping a variable example
+            translated['refs'] = {
+                k: v
+                for k, v in translated['refs'].items()
+                if '/' not in k or k.split('/')[0] in keep
+            }
+            filtered.append(translated)
+        return filtered
+
+    def expand(self, pcoll: beam.PCollection) -> beam.PCollection:
+        return pcoll | beam.Map(self._filter, keep=self.keep)
+
+
+@dataclass
+class ValidateDatasetDimensions(beam.PTransform):
+    """Open the reference.json in xarray and validate dimensions."""
+
+    expected_dims: Dict = field(default_factory=dict)
+
+    @staticmethod
+    def _validate(zarr_store: zarr.storage.FSStore, expected_dims: Dict) -> None:
+        ds = xr.open_dataset(zarr_store, engine='zarr')
+        if set(ds.dims) != expected_dims.keys():
+            raise ValueError(f'Expected dimensions {expected_dims.keys()}, got {ds.dims}')
+        for dim, bounds in expected_dims.items():
+            if bounds is None:
+                continue
+            lo, hi = bounds
+            actual_lo, actual_hi = round(ds[dim].data.min()), round(ds[dim].data.max())
+            if actual_lo != lo or actual_hi != hi:
+                raise ValueError(f'Expected {dim} range [{lo}, {hi}], got {actual_lo, actual_hi}')
+        return ds
+
+    def expand(
+        self,
+        pcoll: beam.PCollection,
+    ) -> beam.PCollection:
+        return pcoll | beam.Map(self._validate, expected_dims=self.expected_dims)
+
+
+recipe = (
+    beam.Create(pattern.items())
+    | OpenWithKerchunk(
+        remote_protocol=earthdata_protocol,
+        file_type=pattern.file_type,
+        # lat/lon are around 5k, this is the best option for forcing kerchunk to inline them
+        inline_threshold=6000,
+        storage_options=fsspec_open_kwargs,
+    )
+    | FilterVars(keep={*pattern.concat_dims, *IDENTICAL_DIMS, *SELECTED_VARS})
+    | WriteCombinedReference(
+        concat_dims=CONCAT_DIMS,
+        identical_dims=IDENTICAL_DIMS,
+        store_name=SHORT_NAME
+    )
+    | ConsolidateMetadata()
+    # | ValidateDatasetDimensions(expected_dims={'time': None, 'lat': (-90, 90), 'lon': (-180, 180)})
```

Comment:
This does not work with the new output of ConsolidateMetadata so will need to revise or remove it.

Reply:
Can you try the local version of ConsolidateMetadata?

```
@beam.ptransform_fn
def ConsolidateMetadata(pcoll: beam.PCollection) -> beam.PCollection:
    """Consolidate metadata into a single .zmetadata field.
    See zarr.consolidate_metadata() for details.
    """

    def _consolidate(store: zarr.storage.FSStore) -> zarr.storage.FSStore:
        zarr.consolidate_metadata(store, path=None)
        return store

    return pcoll | beam.Map(_consolidate)
```

It would be nice if it returned `FSStore` like WriteCombinedReference @norlandrhagen 

@abarciauskas-bgse Why is it preferred to write out the zarr and then consolidate the metadata?

Reply:
@sbquinlan thanks for keeping track of this PR. I think we want to use transforms maintained by pangeo-forge-recipes when possible. In general, I think you have to write the zarr store before running a consolidate metadata operation but perhaps I am misunderstanding your question.

As it happens, we may be removing consolidate metadata altogether since we need to investigate if it is actually helpful to consolidate metadata for kerchunk reference files (or if someone can point me to where this has already been determined, that would be helpful).

Reply:
> I think we want to use transforms maintained by pangeo-forge-recipes when possible

The question is basically should we change the committed implementation of ConsolidateMetadata to return the more useful FSStore instead of the less useful ConsolidatedMetadataStore. I can't see ConsolidatedMetadataStore being useful in pangeo-forge. This is more for @norlandrhagen rather than specific to this PR.

> In general, I think you have to write the zarr store before running a consolidate metadata operation 

Yeah I don't think this is true. It just needs a MutableMapping so you could do it in memory before writing it out, but that's my question basically: is it more performant to consolidate and then write out the references?